### ISSUE TITLE
MongoDB: forced referential actions to prevent looping (target 3.7)

### ISSUE THREAD

**Issue by tanberry** [prisma/prisma2-docs#2629]
Dev work is [PR #2415](https://github.com/prisma/prisma-engines/pull/2415)
Per Flavian: 
Any MongoDB datamodel with self relations or cycles between 3 models will be forced to set some specific referential actions to prevent the referential action emulations from looping infinitely (eg: Cascade cannot work).

The rules and examples have already been all written for SQL Server [here](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions/special-rules-for-referential-actions#self-relation). And [here](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/referential-actions/special-rules-for-referential-actions#cyclic-relation-between-three-tables)

These limitations exist as soon as you enable the `referentialIntegrity = "prisma"` mode. Since it’s enabled for MongoDB by default, it’s the primary target for this change.

Proposal is that we create a different page based on the SQL Server one for the prisma referentialIntegrity mode and there explain how that affects MongoDB.
 
**Note**: The Multiple cascade paths between two models chapter do not apply though.

**Comment by tanberry** [comment]
We will try to merge [another large PR (2581)](https://github.com/prisma/docs/pull/2581) that impacts this same file BEFORE starting work on this ticket, if possible. @janpio do you know if the code side of this emulation work is ready to merge? We are 99% ready on docs side, and as I stated it would be very helpful if we can merge #2581 as soon as possible!

**Comment by janpio** [comment]
I think this has been merged and will be part of 3.7 indeed.

Let's get #2581 over the finish line then to make this easier.

**Comment by janpio** [comment]
(You could also create a PR into the branch of #2581 as a separate thing for this issue - then it is built on the content of that PR already)

**Comment by tanberry** [comment]
Thanks @janpio ! Good news, then @keerlu  will merge  #2581 Monday morning, and I will start a new PR for this work. (Yep, we could have made a PR on top of her PR, but I like the simplicity of first merging hers, then I do a separate one.)

**Comment by tanberry** [comment]
Closing this because it handles the basic problem; will open a new Issue to track larger improvements.
